### DISCUSSION TITLE
[Bug]: Serial Module Configuration > TEXTMSG > Override Console Serial Port appears to be inoperative

### DISCUSSION THREAD

**Discussion by Coopersmith-24601** [meshtastic/firmware#4994] [Troubleshooting]
### Category

Serial

### Hardware

Heltec V3

### Firmware Version

2.4.2.5b45303 

### Description

I have a use case to output the TEXT MESSAGE function of the SERIAL MODULE via the USB-C port.

According to the documentation and video on it I should be able to do that by turning on the Override Console Serial Port switch.

However, when I do that nothing whatsoever appears to be output via the USB-C port.

I know the USB-C port normally is outputting debug and other data in its DEFAULT mode because I can easily see that on the serial monitor I have hooked up to it, but the documentation expressly says I should be able to perform the TEXT MESSAGE function over the USB-C port if the Override Console Serial Port switch is engaged.

**Comment by heyitsyang** [comment]
What am I missing?  I've tried the same thing on both a HeltecV3 and the RAK19007 with no results.  Anything typed in the terminal screen is not echoed back and no message is sent.  Baud rates are all correct and I can verify that the terminal program (PuTTY) is functioning correctly.

**Comment by HabibMCS** [comment]
Any Luck so far on this? I think it is bug. I am also using the heltec v3 and it seems it is not able to recv serial input properly. I am using pins 33 as TX and 34 as RX. According to my recent hit trials observation. Whenever a serial text is sent to the heltec v3 it tries to publish something but it fails due to parsing error. I will try to post the error next time when I see it.

**Comment by Tannerpalin** [comment]
Unfortunately I'm still running into this on my Heltec v3 using Pins 43 and 44 coming from the CP2102 USB to UART converter. We should be able to override these unlock serial messages over this UART port. This would be extremely useful.

**Comment by DHaacke** [comment]
I am having the same trouble with my Heltec V3 using pins 44 (RX) and 43 (TX), but the Heltec doesn't even try to send anything. In fairness to HelTec and my lack of expertise, I haven't set up any type of debugging. If anyone knows a way, using a Mac, to debug that the sending serial port is actually sending data, I'd sure appreciate knowing. Thanks in advance, and holler if you guys have any success.

**Comment by spechter** [comment]
I've found some information about the pins on JP2 on [Heltec's wiki](https://wiki.heltec.org/docs/devices/open-source-hardware/esp32-series/lora-32/wifi-lora-32-v3/Pin-diagram-guidance#recommended-gpios-for-external-hardware-beginner-friendly); basically half of the pins on that jumper are reserved up until pin 47 and 48.

Setting 47/48 in the android client as TX/RX and mode as 'simple' I was able to drum some data into my node. 

The console on the USB port keeps providing debug output.

This in contrast to using pin 43/44 which are the pins used by the USB-UART converter servicing the USB port on the device.

Be advised that data transmitted in 'simple' mode is not shown in the Android client but requires a second node to be configured with serial communication as well.
